clients/src/main/java/org/apache/kafka/common/metrics/Gauge.java

```diff
@@ -20,13 +20,4 @@
  * A gauge metric is an instantaneous reading of a particular value.
  */
 @FunctionalInterface
-public interface Gauge<T> extends MetricValueProvider<T> {
-
-    /**
-     * Returns the current value associated with this gauge.
-     * @param config The configuration for this metric
-     * @param now The POSIX time in milliseconds the measurement is being taken
-     */
-    T value(MetricConfig config, long now);
-
-}
+public interface Gauge<T> extends MetricValueProvider<T> { }
```

Comment:
The `Gauge` type seems redundant at this stage. Would it be worth filing a KIP to deprecate it?